pulsar-client-cpp/lib/Commands.cc

```diff
@@ -23,10 +23,15 @@
 #include "LogUtils.h"
 #include "checksum/ChecksumProvider.h"
 #include <algorithm>
+#include <boost/thread/mutex.hpp>
+
 namespace pulsar {
 
 using namespace pulsar::proto;
 
+static BaseCommand cmd;
+static boost::mutex mutex;
```

Comment:
Last comment (I promise). I was meaning for the 2 static objects to be declared directly in their methods. Eg:

```cpp
SharedBuffer Commands::newPartitionMetadataRequest(const std::string& topic, uint64_t requestId) {
    static BaseCommand cmd;
    static boost::mutex mutex;

     mutex.lock();
    cmd....
    ....
```

That makes more clear the scope in which the `cmd` and `mutex` are used.